src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -428,9 +424,16 @@ protected virtual void Initialize()
 
                 LoadCustomExtensions();
 
+                // Do this after we've loaded the custom extensions. That gives an extension an opportunity to plug in their own implementations.
+                if (storageString != null)
+                {
+                    var lockManager = this.ScriptConfig.HostConfig.GetService<IDistributedLockManager>();
```

Comment:
Really wish we were not committing further to the pattern of pulling stuff from the service provider directly like this. We'll need to rework how DI is done, so if giving extensions the ability to inject an `IDistributedLockManager` is just a nice to have, let's avoid this here so we can have it properly done when we get to it.

Also, is the core SDK registering a default implementation of a `IDistributedLockManager` implementation before this point here? Replacing services in different stages in this process makes it easy to introduce bugs like what we had with configuration and logging. If we want extensions to participate in DI, we should have that properly exposed and allow them to do so in the right place.